mlflow/xgboost.py

```diff
@@ -416,10 +437,36 @@ def log_feature_importance_plot(features, importance, importance_type):
             finally:
                 shutil.rmtree(tmpdir)
 
-        try_mlflow_log(log_model, model, artifact_path="model")
+        # dtrain must exist as the original train function already ran successfully
+        dtrain = args[1] if len(args) > 1 else kwargs["dtrain"]
+
+        input_example = None
+        signature = None
+        try:
+            input_example = dtrain.data_copy
+
+            # it is possible that the dataset was constructed before the patched
+            #   constructor was applied, so we cannot assume the data_copy exists
+            if input_example is None:
+                raise Exception("failed to gather example input.")
```

Comment:
Can we provide a more specific / actionable message to the user? E.g., `"please ensure that autologging is enabled before you import or construct your training dataset"`?